connect-inject/container_init.go

```diff
@@ -202,4 +242,16 @@ consul-k8s connect-init -pod-name=${POD_NAME} -pod-namespace=${POD_NAMESPACE} \
   {{- if .ConsulNamespace }}
   -namespace="{{ .ConsulNamespace }}" \
   {{- end }}
-  -bootstrap > /consul/connect-inject/envoy-bootstrap.yaml`
+  -bootstrap > /consul/connect-inject/envoy-bootstrap.yaml
+
+{{- if .EnableTransparentProxy }}
+
+# Apply traffic redirection rules.
+/consul/connect-inject/consul connect redirect-traffic \
+  {{- if .ConsulNamespace }}
+  -namespace="{{ .ConsulNamespace }}" \
+  {{- end }}
+  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
+  -proxy-uid=0
```

Comment:
If someone changes `rootUserAndGroupID` will this also need to be updated? If so, maybe this should be added as a templated field.

Reply:
That one is only used by the init container, where is `-proxy-uid` is the user id of the Envoy proxy container. Once we don't run as root, this will need to be changed. I think we can add it to the template once we support running envoy as non-root.

Note that to support tproxy and be able to run iptables we need that init container to be root, unfortunately.